### PR TITLE
Simplify output shape inference for dot/cos merge

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -248,7 +248,7 @@ def batch_dot(x, y, axes=None):
         axes: list (or single) int with target dimensions
 
     # Returns
-        Tensor with ndim >= 2
+        A tensor with shape equal to the concatenation of x's shape (less the dimension that was summed over) and y's shape (less the batch dimension and the dimension that was summed over). If the final rank is 1, we reshape it to (batch_size, 1).
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: A couple examples please!

**Comment by fchollet** [comment]
LGTM otherwise.

**Comment by fchollet** [comment]
Please avoid pushing many commits sequentially, group them together, otherwise there is a large burden on Travis: travis

**Comment by farizrahman4u** [comment]
Oops! My bad. Was under the assumption that older builds will be terminated when new commits came in.

**Comment by fchollet** [comment]
> Was under the assumption that older builds will be terminated when new commits came in.

That would prevent you from knowing the specific commit when a bug was introduced. In our case, tests are heavy so it's simply not possible to run tests after every commit. Running them once per PR should be enough.

**Comment by farizrahman4u** [comment]
@fchollet  Is the example ok?

**Comment by fchollet** [comment]
LGTM

**Pull request merged**